proxies/go/handler_command.go

```diff
@@ -228,6 +226,9 @@ func callMethodOnEntity(
 	locationId := strconv.Itoa(len(datastore.commandResults[command]))
 	locationHeader := "command/" + command + "/" + locationId
 
+	commandMutex.Lock()
+	defer commandMutex.Unlock()
```

Comment:
I think you still need to lock around the reads as well.

Since maps are pointer types, might be able to make a `getMap` method that does the locking and returns a reference to it, then use that everywhere instead?

Reply:
Ignore that idea - just remembered that you won't be able to defer the unlock that way.

I think we still need at least a `sync.RWMutex` for every access to the map though.